src/libs/Navigation/AppNavigator/createCustomStackNavigator/CustomRouter.ts

```diff
@@ -97,6 +100,23 @@ function compareAndAdaptState(state: StackNavigationState<RootStackParamList>) {
     }
 }
 
+function shouldPreventReset(state: StackNavigationState<ParamListBase>, action: CommonActions.Action | StackActionType) {
+    if (action.type !== CONST.NAVIGATION_ACTIONS.RESET || !action?.payload) {
+        return false;
+    }
+    const currentFocusedRoute = findFocusedRoute(state);
+    const targetFocusedRoute = findFocusedRoute(action?.payload);
+
+    // We want to prevent the user from navigating back to a non-onboarding screen if they are currently on an onboarding screen
+    if (isOnboardingFlowName(currentFocusedRoute?.name) && !isOnboardingFlowName(targetFocusedRoute?.name)) {
+        Welcome.setOnboardingErrorMessage(Localize.translateLocal('onboarding.purpose.errorBackButton'));
+        // We reset the URL as the browser sets it in a way that doesn't match the navigation state
+        // eslint-disable-next-line no-restricted-globals
+        history.replaceState({}, '', getPathFromState(state, linkingConfig.config));
```

Comment:
`history` does not exist on native. I am not sure if a real user could land into this code but I got a crash while testing another PR

<img width="480" alt="Screenshot 2024-10-09 at 8 07 01 PM" src="https://github.com/user-attachments/assets/e139c73e-c10f-4f38-8386-6952bfd9315a">

Reply:
Yeah, we definetly should make this part platform-specific.

Reply:
Somebody from SWM can take care of this issue if you want

Reply:
https://github.com/Expensify/App/issues/50713